users/service.go

```diff
@@ -73,11 +75,13 @@ func (svc usersService) Register(ctx context.Context, user User) error {
 func (svc usersService) Login(ctx context.Context, user User) (string, error) {
 	dbUser, err := svc.users.RetrieveByID(ctx, user.Email)
 	if err != nil {
-		return "", ErrUnauthorizedAccess
+		// return "", ErrUnauthorizedAccess
+		return "", fmt.Errorf("%w: %s", ErrUnauthorizedAccess, err)
```

Comment:
Here you wrap, but somewhere you need to log as well (print for our server logs, not send to client). I guess you do it in `encodeError()`, but I do not see logger priniting there.

Reply:
Whole error, with all wrapped layers, will go through logger and every of wrapped errors will be displayed in logs. On the other hand, client will see only last (outer) wrapper